rtl/eclrtl/eclregex.cpp

```diff
@@ -376,12 +419,11 @@ class CCompiledStrRegExpr : implements ICompiledStrRegExpr
 
     void replace(size32_t & outlen, char * & out, size32_t slen, char const * str, size32_t rlen, char const * replace) const
     {
-        PCRE2_SIZE pcreLen = 0;
         outlen = 0;
-        pcre2_match_data_8 * matchData = pcre2_match_data_create_from_pattern_8(compiledRegex.get(), pcre2GeneralContext8);
+        PCRE2MatchData8 matchData = pcre2_match_data_create_from_pattern_8(compiledRegex.get(), pcre2GeneralContext8);
```

Comment:
If the copy constructor is removed this may give an error.  Strictly speaking we should use:

```
PCRE2MatchData8 matchData(pcre2_match_data_create_from_pattern_8(compiledRegex.get(), pcre2GeneralContext8));
```
but we don't follow that pattern in so many places (and it is optimized by the compiler)...